Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -0,0 +1,512 @@
+    private var renderVideoFeed = true
+    
+    /// Used to determine the device location when originCamera is not set.
+    private lazy var locationManager: CLLocationManager = { [unowned self] in
+        let lm = CLLocationManager()
+        lm.desiredAccuracy = kCLLocationAccuracyBest
+        lm.delegate = self
+        return lm
+    }()
+    
+    /// Initial location from locationManager.
+    private var initialLocation: CLLocation?
+    
+    /// Current horizontal accuracy of the device.
+    private var horizontalAccuracy: CLLocationAccuracy = .greatestFiniteMagnitude
+    
+    /// The intial camera position and orientation whether it was set via originCamera or the locationManager.
+    private var initialTransformationMatrix = AGSTransformationMatrix()
+    
+    /// Whether `ARKit` is supported on this device.
+    private var isSupported = {
+        return ARWorldTrackingConfiguration.isSupported
+    }()
+    
+    /// Whether the client has been notfiied of start/failure.
+    private var notifiedStartOrFailure = false
+    
+    /// Used when calculating framerate.
+    private var lastUpdateTime: TimeInterval = 0
+    
+    // A quaternion used to compensate for the pitch beeing 90 degrees on `ARKit`; used to calculate the current device transformation for each frame.
+    let compensationQuat:simd_quatd = simd_quatd(ix: (sin(45 / (180 / .pi))), iy: 0, iz: 0, r: (cos(45 / (180 / .pi))))
+    
+    /// The quaternion used to represent the device orientation; used to calculate the current device transformation for each frame; defaults to landcape-left.
+    var orientationQuat:simd_quatd = simd_quatd(statusBarOrientation: .landscapeLeft)
+    
+    // MARK: Initializers
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        sharedInitialization()
+    }
+    
+    required public init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        sharedInitialization()
+    }
+    
+    /// Initializer used to denote whether to display the live camera image.
+    ///
+    /// - Parameter renderVideoFeed: Whether to display the live camera image.
+    public convenience init(renderVideoFeed: Bool){
+        self.init(frame: .zero)
+        self.renderVideoFeed = renderVideoFeed
+    }
+    
+    deinit {
+        stopTracking()
+    }
+    
+    /// Initialization code shared between all initializers.
+    private func sharedInitialization(){
+        // Add the ARSCNView to our view.
+        addSubviewWithConstraints(arSCNView)
+        arSCNView.delegate = self
+        
+        // Add sceneView to view and setup constraints.
+        addSubviewWithConstraints(sceneView)
+
+        // Make our sceneView's background transparent, no atmosphereEffect.
+        sceneView.isBackgroundTransparent = true
+        sceneView.atmosphereEffect = .none
+        
+        // Tell the sceneView we will be calling `renderFrame()` manually.
+        sceneView.isManualRendering = true
```

Comment:
do you want to do this even if not showing the video feed from the camera?

Reply:
I'm not sure.  I am going to enter an issue in our repo so the team can talk about it.  That OK?

Reply:
ok